plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -426,5 +442,37 @@ def temp_release_docs
         end
       end
     end
+
+    def finish_blue_green_deployment(release, release_docs)
+      switch_blue_green_service(release_docs)
+      previous = release.previous_successful_release
+      if previous && previous.blue_green_color != release.blue_green_color
+        previous.release_docs.each { |d| delete_blue_green_resources(d) }
+      end
+    end
+
+    def switch_blue_green_service(release_docs)
+      release_docs.each do |release_doc|
+        next unless service = service_resource(release_doc)
+        @output.puts "Switching service for #{release_doc.deploy_group.name} " \
+          "role #{release_doc.kubernetes_role.name} to #{release_doc.blue_green_color.upcase}"
+        service.deploy
+      end
+    end
+
+    def delete_blue_green_resources(release_doc)
+      puts_action "Deleting", release_doc
+      non_service_resources(release_doc).each(&:delete)
+    end
+
+    # used for blue_green deployment
+    def non_service_resources(release_doc)
+      release_doc.resources - Array(service_resource(release_doc))
+    end
+
+    # used for blue_green deployment
+    def service_resource(release_doc)
+      release_doc.resources.detect { |r| r.is_a?(Kubernetes::Resource::Service) }
```

Comment:
Looking again to it I'm wondering if there would be some case with more than one service in the same release_doc 🤔 If so, maybe we could use `select` instead of `detect` and apply the switch for all services in then list?

Reply:
yeah, let's do that ... was thinking about that too